src/store/stateSlice.ts

```diff
@@ -0,0 +1,128 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { IPersonCardProps } from 'components/personCard';
+import { IFormValues } from 'components/formTypes';
+import {
+  IItemData,
+  SORTING,
+  DEFAULT_PER_PAGE,
+  IGetItemsOptions,
+  getItems,
+  IItemsResp,
+} from 'api/api';
+
+interface IItemsPayload {
+  itemsData: IItemData[];
+  errMsg: string;
+  pages: number;
+  search?: string;
+  sorting?: SORTING;
+  page?: number;
+  itemsPerPage?: number;
+}
+
+export type FormValuesState = Omit<IFormValues, 'picture'>;
+
+export interface IState {
+  personCards: IPersonCardProps[];
+  formValues?: FormValuesState;
+  itemsData?: IItemData[];
+  isDataLoading: boolean;
+  errMsg: string;
+  search: string;
+  pages: number;
+  sorting: SORTING;
+  itemsPerPage: number;
+  page: number;
+  detailsIdx: number;
+}
+
+export const initialState: IState = {
+  personCards: [],
+  isDataLoading: true,
+  errMsg: '',
+  search: '',
+  pages: 1,
+  sorting: SORTING.nameAsc,
+  itemsPerPage: DEFAULT_PER_PAGE,
+  page: 1,
+  detailsIdx: -1,
+};
+
+export interface IGetItemsThunkData extends IItemsResp {
+  search: string;
+  sorting: SORTING;
+  page: number;
+  itemsPerPage: number;
+}
+
+export const getItemsThunk = createAsyncThunk<
+  IGetItemsThunkData,
+  IGetItemsOptions,
+  { rejectValue: string }
+>('state/getItemsThunk', async function (opt) {
+  const resp = await getItems(opt);
+
+  return {
+    items: resp.items,
+    pages: resp.pages,
+    errMsg: resp.errMsg,
+    search: opt.search,
+    sorting: opt.sorting,
+    page: opt.page,
+    itemsPerPage: opt.limit,
+  };
+});
+
+const stateSlice = createSlice({
```

Comment:
The idea of "slices" is to split your app state into some independent parts of the state. For example, your form data does not care about your page sorting. At least there should be two slices for the page with forms and for the page with search bar